File: shared/src/main/java/org/tiqr/authenticator/auth/EnrollmentChallenge.java
package org.tiqr.authenticator.auth;

import android.os.Parcel;

/**
 * Represents an enrollment challenge.
 */
public class EnrollmentChallenge extends Challenge {
    private String _enrollmentURL;


    /**
     * Factory.
     */
    public static final Creator<EnrollmentChallenge> CREATOR = new Creator<EnrollmentChallenge>() {
        public EnrollmentChallenge createFromParcel(Parcel source) {
            return new EnrollmentChallenge(source);
        }

        public EnrollmentChallenge[] newArray(int size) {
            return new EnrollmentChallenge[size];
        }
    };

    /**
     * Constructor.
     */
    public EnrollmentChallenge() {
    }

    /**
     * Constructor.
     */
    private EnrollmentChallenge(Parcel source) {
        super(source);
        _enrollmentURL = source.readString();
    }

    /**
     * Export to parcel.
     */
    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
        dest.writeString(_enrollmentURL);
    }

    /**
     * Enrollment callback URL.
     * 
     * @return enrollment callback URL.
     */
    public String getEnrollmentURL() {
        return _enrollmentURL;
    }

    /**
     * Sets the enrollment callback URL for this challenge.
     */
    public void setEnrollmentURL(String enrollmentURL)
    {
        _enrollmentURL = enrollmentURL;
    }
}

File: shared/src/main/java/org/tiqr/authenticator/auth/EnrollmentChallenge.kt
package org.tiqr.authenticator.auth

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import org.tiqr.authenticator.datamodel.Identity
import org.tiqr.authenticator.datamodel.IdentityProvider

/**
 * Represents an enrollment challenge.
 */
@Parcelize
class EnrollmentChallenge(

        val enrollmentURL: String,

        override val protocolVersion: String?,
        override val identityProvider: IdentityProvider,
        override var identity: Identity,
        override val returnURL: String?

) : Challenge(protocolVersion, identityProvider, identity, returnURL), Parcelable